src/renderers/dom/server/ReactServerUpdateQueue.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactServerUpdateQueue
+ * @flow
+ */
+
+'use strict';
+
+var ReactUpdateQueue = require('ReactUpdateQueue');
+var Transaction = require('Transaction');
+var ReactClass = require('ReactClass');
+var warning = require('warning');
+
+function warnTDZ(publicInstance: ReactClass, callerName: string) {
+  if (__DEV__) {
+    warning(
+      false,
+      '%s(...): Can only update a mounting component. ' +
+      'This usually means you called %s() outside componentWillMount() on the server. ' +
+      'This is a no-op. Please check the code for the %s component.',
+      callerName,
+      callerName,
+      publicInstance.constructor.displayName || publicInstance.constructor.name || 'ReactClass'
```

Comment:
Presumably `publicInstance.constructor` check still had some merit—sorry I didn’t include it in my comment.
`{ render: ... }` is technically a valid React component so let’s make sure we don’t crash for it.